lib/smart_proxy_container_gateway/container_gateway_api.rb

```diff
@@ -61,6 +68,24 @@ class Api < ::Sinatra::Base
         { repositories: Proxy::ContainerGateway.unauthenticated_repos }.to_json
       end
 
+      get '/v2/token' do
+        response.headers['Docker-Distribution-API-Version'] = 'registry/2.0'
+
+        unless auth_header.present? && auth_header.basic_auth?
+          return { token: AuthorizationHeader::UNAUTHORIZED_TOKEN, issued_at: Time.now, expires_at: Time.now + 60 }.to_json
```

Comment:
What do you think about making the time configurable?

Reply:
for the unauthenticated token I'm not sure it really matters, we could probably just return it for 1 year in the future (in retrospect 60 might be too short).  

However i'm not even sure the client stores this, so in the end it may not matter at all

Reply:
Oh never mind, I was thinking about the token coming from Katello.

Reply:
updated to 1 year